cr-examples/samples/src/main/java/oracle/code/samples/DialectSample.java

```diff
@@ -0,0 +1,281 @@
+
+        // 3. Print the original code model
+        String codeModelString = functionModel.toText();
+        System.out.println(codeModelString);
+
+        // 3. Transform the code model to include our custom Op
+        CoreOp.FuncOp dialectModel = functionModel.transform((builder, op) -> {
+            CopyContext context = builder.context();
+            // 4 Find the JavaOp.Op node
+            if (op instanceof JavaOp.AddOp addOp) {
+                // 5. Obtain the operands for the node
+                List<Value> inputOperands = addOp.operands();
+                List<Value> outputOperands = context.getValues(inputOperands);
+
+                // 6. Create a new Op with the new operation
+                MyAdd myAddOp = new MyAdd("myAdd", outputOperands, JavaType.INT);
+
+                // 7. Attach the new Op to the builder
+                Op.Result result = builder.op(myAddOp);
+
+                // 8. Propagate the location of the new op
+                myAddOp.setLocation(addOp.location());
+
+                // 9. Map the values from input -> output for the new Op
+                context.mapValue(addOp.result(), result);
+            } else {
+                builder.op(op);
+            }
+            return builder;
+        });
+
+        // 10. Print the transformed code model
+        System.out.println("Model with new OpNodes for Dialect: ");
+        System.out.println(dialectModel.toText());
+
+        System.out.println("Print Code Tree: ");
+        printCodeTree(dialectModel);
+
+        // Currently, we can't interpreter a code model with dialect ops
+        //var result = Interpreter.invoke(MethodHandles.lookup(), dialectModel,  10, 20);
+        //System.out.println("Result: ");
+    }
+
+
+    // Part B: Create a new Op for specializing the invoke Op.
+    private static int myIntrinsic(int a, int b) {
+        return a + b;
+    }
+
+    @CodeReflection
+    public static int myFunction(int a, int b) {
+        return myIntrinsic(a, b);
+    }
+
+    public static class MyInvoke extends Op { // externalized
+
+        private final TypeElement typeDescriptor;
+
+        MyInvoke(String opName, TypeElement typeDescriptor, List<Value> operands) {
+            super(opName, operands);
+            this.typeDescriptor = typeDescriptor;
+        }
+
+        MyInvoke(Op that, CopyContext cc) {
+            super(that, cc);
+            this.typeDescriptor = that.resultType();
+        }
+
+        @Override
+        public Op transform(CopyContext copyContext, OpTransformer opTransformer) {
+            return new MyInvoke(this, copyContext);
+        }
+
+        @Override
+        public TypeElement resultType() {
+            return typeDescriptor;
+        }
+
+        @Override
+        public List<Value> operands() {
+            return super.operands();
+        }
+
+        @Override
+        public Map<String, Object> externalize() {
+            return Map.of("", this.typeDescriptor);
```

Comment:
This can be the name of the function to invoke, which you are currently using as the operation name.